### PR TITLE
[Concept Entry] AI Simulated Annealing #2019

### PR DESCRIPTION
**Pull request by DevGat718**
### Description

 created an entry for simulated annealing (AI)

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [x ] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:1]: Remove this file from this PR. It looks like you made a branch off your earlier branch for your Neural Network PR rather than a fresh branch from main.

File: documentation/tags.md
Comment (SSwiniarski) [inline documentation/tags.md:1]: Undo these changes, as they're leftover from the prior branch.

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Tags:
+  - 'Search Algorithms'
+  - 'Machine Learning'
+  - 'AI'
+CatalogContent:
+  - 'paths/data-science'
+  - 'paths/machine-learning-ai-engineering-foundations'
+---
+
+**Simulated Annealing** An optimization algorithm, used to solve an optimization problems which can contain many different solutions (by the process of slowly decreasing the probability of accepting worse solutions as the vast solution space is explored). Each solution has a different score & the goal of the algorithm is to find the best score over the course of the experiment.
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:17]:
```suggestion
**Simulated Annealing** An optimization algorithm, used to solve an optimization problems which can contain many different solutions by the process of slowly decreasing the probability of accepting worse solutions as the vast solution space is explored. Each solution has a different score and the goal of the algorithm is to find the best score over the course of the search.
```

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Describe The Method Of Simulated Annealing
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:19]:
```suggestion
## The Method Of Simulated Annealing

```

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- Simulated annealing creates a trial point randomly then the algorithm selects the distance between the current point and the trial point by a probability distribution.
+- The Simulated Annealing formula then determines if the new point is better than the older point or not. If the new point is better, it is made as a next point, but if the new point is worse, it can still be accepted depending on the acceptance function.
+- The annealing parameters are set, raising and reducing the temperatures (the scale of such distribution from point to point) for lowering values. The simulated annealing is concluded when it reaches the lowest minima or hits a certain stopping criteria.
+
+## Describe It's Relation To AI
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:24]:
```suggestion
## It's Relation To AI

```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@DevGat718 I did a first review and left a few comments. Particularly you need to remove the changes left by the prior branch so the changes in this PR are limited to your new entry.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:1]: done

File: documentation/tags.md
Comment (DevGat718) [reply documentation/tags.md:1]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md (hunk 1 above)
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:17]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md (hunk 2 above)
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:19]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md (hunk 3 above)
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:24]: done

**Comment by DevGat718** [comment]
I'm caught up. @SSwiniarski

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:1]: my bad

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/neural-networks.md:1]: Ok, now you need to put this file back. It was the one at the other location that needed to be deleted,

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@DevGat718 Ok, I think some wires got crossed. You ended up deleting the file that was merged. That's not what we want to do. Can you restore neural networks.

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/neural-networks.md:1]: How would I add it back? 
So if I wasn't suppose to take that out what was I suppose to take out then?

**Comment by DevGat718** [comment]
I got it back

**Comment by SSwiniarski** [comment]
@DevGat718 ok, I deleted the correct file for you.

**Review by SSwiniarski — APPROVED** [review]
@DevGat718 looks good for a second review.

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## The Method Of Simulated Annealing
+- Simulated annealing creates a trial point randomly then the algorithm selects the distance between the current point and the trial point by a probability distribution.
+- The Simulated Annealing formula then determines if the new point is better than the older point or not. If the new point is better, it is made as a next point, but if the new point is worse, it can still be accepted depending on the acceptance function.
+- The annealing parameters are set, raising and reducing the temperatures (the scale of such distribution from point to point) for lowering values. The simulated annealing is concluded when it reaches the lowest minima or hits a certain stopping criteria.
+
+## It's Relation To AI
+- Simulated annealing uses machine learning to constantly improve itself based on an algorithm which tests various points over and over to find the lowest minima. This is done via a similar process called physical annealing, the process through which a material is heated till the annealing temperature and then cooled down for the desired structure formation.
```
Comment (caupolicandiaz) [inline content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:25]:
```suggestion
- Simulated annealing is a probabilistic algorithm which tests points across a solution space to find the lowest minima. The algorithm is termed "simulated annealing" because it mirrors the process of physical annealing, a process in which a material is repeatedly heated and cooled to elicit desired structural properties.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@DevGat718 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  - 'Machine Learning'
+  - 'AI'
+CatalogContent:
+  - 'paths/data-science'
+  - 'paths/machine-learning-ai-engineering-foundations'
+---
+
+**Simulated Annealing** An optimization algorithm, used to solve an optimization problems which can contain many different solutions by the process of slowly decreasing the probability of accepting worse solutions as the vast solution space is explored. Each solution has a different score and the goal of the algorithm is to find the best score over the course of the search.
+
+## The Method Of Simulated Annealing
```
[the remark this answers is not in the document]
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:19]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- Simulated annealing creates a trial point randomly then the algorithm selects the distance between the current point and the trial point by a probability distribution.
+- The Simulated Annealing formula then determines if the new point is better than the older point or not. If the new point is better, it is made as a next point, but if the new point is worse, it can still be accepted depending on the acceptance function.
+- The annealing parameters are set, raising and reducing the temperatures (the scale of such distribution from point to point) for lowering values. The simulated annealing is concluded when it reaches the lowest minima or hits a certain stopping criteria.
+
+## It's Relation To AI
```
[the remark this answers is not in the document]
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:24]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+---
+Title: 'Simulated Annealing'
+Description: 'An optimization algorithm, used to solve an optimization problems which can contain many different solutions. Each solution has a different score & the goal of the algorithm is to find the best score.'
```
[the remark this answers is not in the document]
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:3]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  - 'AI'
+CatalogContent:
+  - 'paths/data-science'
+  - 'paths/machine-learning-ai-engineering-foundations'
+---
+
+**Simulated Annealing** An optimization algorithm, used to solve an optimization problems which can contain many different solutions by the process of slowly decreasing the probability of accepting worse solutions as the vast solution space is explored. Each solution has a different score and the goal of the algorithm is to find the best score over the course of the search.
+
+## The Method Of Simulated Annealing
+- Simulated annealing creates a trial point randomly then the algorithm selects the distance between the current point and the trial point by a probability distribution.
```
[the remark this answers is not in the document]
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:20]: done

**Comment by DevGat718** [comment]
sorry was away for work training, im back!

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+- The Simulated Annealing formula then determines if the new point is better than the older point or not. If the new point is better, it is made as a next point, but if the new point is worse, it can still be accepted depending on the acceptance function.
```
[the remark this answers is not in the document]
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:21]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+Tags:
+  - 'Search Algorithms'
+  - 'Machine Learning'
+  - 'AI'
+CatalogContent:
+  - 'paths/data-science'
+  - 'paths/machine-learning-ai-engineering-foundations'
+---
+
+**Simulated Annealing** An optimization algorithm, used to solve an optimization problems which can contain many different solutions by the process of slowly decreasing the probability of accepting worse solutions as the vast solution space is explored. Each solution has a different score and the goal of the algorithm is to find the best score over the course of the search.
```
[the remark this answers is not in the document]
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:17]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+## The Method Of Simulated Annealing
+- Simulated annealing creates a trial point randomly then the algorithm selects the distance between the current point and the trial point by a probability distribution.
+- The Simulated Annealing formula then determines if the new point is better than the older point or not. If the new point is better, it is made as a next point, but if the new point is worse, it can still be accepted depending on the acceptance function.
+- The annealing parameters are set, raising and reducing the temperatures (the scale of such distribution from point to point) for lowering values. The simulated annealing is concluded when it reaches the lowest minima or hits a certain stopping criteria.
```
[the remark this answers is not in the document]
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:22]: done

File: content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md (hunk 4 above)
Comment (DevGat718) [reply content/ai/concepts/search-algorithms/terms/simulated-annealing/simulated-annealing.md:25]: done

**Review by caupolicandiaz — APPROVED** [review]
@DevGat718 Looks good. I just made a few small edits to adjust the spacing and tweak some of the language I suggested in the previous round.

**Comment by SSwiniarski** [comment]
@DevGat718 Congrats on the new entry. Live link here: https://www.codecademy.com/resources/docs/ai/search-algorithms/simulated-annealing

image

**Comment by DevGat718** [comment]
Yay! thank you.
